clouddriver-ecs/src/integration/java/com/netflix/spinnaker/clouddriver/ecs/test/CreateServerGroupWithArtifactsSpec.java

```diff
@@ -0,0 +1,311 @@
+
+    ArgumentCaptor<CreateServiceRequest> createServiceArgs =
+        ArgumentCaptor.forClass(CreateServiceRequest.class);
+    verify(mockECS).createService(createServiceArgs.capture());
+    CreateServiceRequest seenCreateServRequest = createServiceArgs.getValue();
+    assertEquals("EC2", seenCreateServRequest.getLaunchType());
+    assertEquals(expectedServerGroupName, seenCreateServRequest.getServiceName());
+    assertEquals(1, seenCreateServRequest.getLoadBalancers().size());
+    LoadBalancer serviceLB = seenCreateServRequest.getLoadBalancers().get(0);
+    assertEquals("application", serviceLB.getContainerName());
+    assertEquals(80, serviceLB.getContainerPort().intValue());
+    assertEquals("integArtifactEC2TgMappings-cluster", seenCreateServRequest.getCluster());
+  }
+
+  @DisplayName(
+      ".\n===\n"
+          + "Given description w/ task def artifacts, FARGATE  launch type, and new target group fields, "
+          + "successfully submit createServerGroup operation"
+          + "\n===")
+  @Test
+  public void createSGOWithArtifactsFARGATETargetGroupMappingsTest() throws IOException {
+
+    // given
+    String url = getTestUrl(CREATE_SG_TEST_PATH);
+    String requestBody =
+        generateStringFromTestFile("/createServerGroup-artifact-FARGATE-targetGroupMappings.json");
+    String expectedServerGroupName = "ecs-integArtifactsFargateTgMappingsStack-detailTest-v000";
+
+    setEcsAccountCreds();
+
+    ByteArrayInputStream byteArrayInputStreamOfArtifactsForFargateType =
+        new ByteArrayInputStream(
+            generateStringFromTestArtifactFile(
+                    "/createServerGroup-artifact-Fargate-targetGroup-artifactFile.json")
+                .getBytes());
+
+    when(artifactDownloader.download(any(Artifact.class)))
+        .thenReturn(byteArrayInputStreamOfArtifactsForFargateType);
+
+    String taskId =
+        given()
+            .contentType(ContentType.JSON)
+            .body(requestBody)
+            .when()
+            .post(url)
+            .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
+            .body("id", notNullValue())
+            .body("resourceUri", containsString("/task/"))
+            .extract()
+            .path("id");
+
+    try {
+      retryUntilTrue(
+          () -> {
+            List<Object> taskHistory =
+                get(getTestUrl("/task/" + taskId))
+                    .then()
+                    .contentType(ContentType.JSON)
+                    .extract()
+                    .path("history");
+            if (taskHistory
+                .toString()
+                .contains(String.format("Done creating 1 of %s", expectedServerGroupName))) {
+              return true;
+            }
+            return false;
+          },
+          String.format("Failed to detect service creation in %s seconds", TASK_RETRY_SECONDS),
+          TASK_RETRY_SECONDS);
+    } catch (InterruptedException e) {
```

Comment:
I'd prefer we not catch this but instead let it bubble up/fail, since we can't proceed with validating the test if it occurs. 
(NOTE: did eclipse auto-generate this block for you? it can't always be trusted.)

Reply:
It relates to my comment on your PR where I had asked why have you used `**throws InterruptedException**` for the tests. For me, eclipse auto-generated this try-catch but even I would also prefer using `** throws InterruptedException**` for my tests. I will make that change.